src/app/clients/ActionBarModal.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState } from "react";
+import styled from "styled-components";
+import Button from "@mui/material/Button/Button";
+import Modal from "@/components/Modal/Modal";
+import { DatePicker, TimePicker } from "@mui/x-date-pickers";
+import dayjs, { Dayjs } from "dayjs";
+import { ClientsTableRow } from "@/app/clients/getClientsTableData";
+
+interface SharedModalProps {
+    isOpen: boolean;
+    onClose: () => void;
+    data: ClientsTableRow[];
+    status: string | null;
+    onSubmit: (date: Dayjs) => void;
+    header: string;
+    headerId: string;
+    errorText: string | null;
+}
+
+const Row = styled.div`
+    display: flex;
+    gap: 1rem;
+    align-items: center;
+`;
+
+const ModalInner = styled.div`
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+    align-items: stretch;
+`;
+
+// TODO VFB-16 Make these border colours change on dark and light themes
+const StatusText = styled.p`
+    margin-left: 1rem;
+    border-top: 1px solid darkgrey;
+    padding: 1rem 0;
+    &:last-child {
+        border-bottom: 1px solid darkgrey;
+    }
+`;
+
+const ActionBarModal: React.FC<SharedModalProps> = (props) => {
+    const [date, setDate] = useState(dayjs(Date()));
+
+    useEffect(() => {
+        setDate(dayjs(Date()));
+    }, [props.isOpen]);
+
+    const onDateChange = (newDate: Dayjs | null): void =>
+        setDate((date) =>
+            date
+                .set("year", newDate?.year() ?? date.year())
+                .set("month", newDate?.month() ?? date.month())
+                .set("day", newDate?.day() ?? date.day())
+        );
+
+    const onTimeChange = (newDate: Dayjs | null): void =>
+        setDate((date) =>
+            date
+                .set("hour", newDate?.hour() ?? date.hour())
+                .set("minute", newDate?.minute() ?? date.minute())
+        );
+
+    return (
+        <Modal {...props} header={props.header} headerId={props.headerId}>
+            <ModalInner>
+                <Row>
+                    <p>Date: </p>
+                    <DatePicker
+                        value={date}
+                        defaultValue={date}
+                        onChange={onDateChange}
+                        disableFuture
+                    />
+                </Row>
+                <Row>
+                    <p>Time: </p>
+                    <TimePicker value={date} onChange={onTimeChange} disableFuture />
+                </Row>
+                <p>Applying:</p>
+                <div>
+                    <StatusText key={props.status}>{props.status}</StatusText>
+                </div>
+                <p>To</p>
+                <div>
```

Comment:
what's the difference between a `<div></div>` and a `<></>`?

Reply:
By having it as an element, it affects the flex layout (specifically where the flex gap appears)